src/replication.c

```diff
@@ -936,16 +873,13 @@ update_replica_entry(spec_t *spec, replica_t *replica, int iofd, char *replicaip
 	write(replica->iofd, spec->lu->volname, strlen(spec->lu->volname));
 	free(rio);
 
-	update_volstate(spec);
-
 	event.data.fd = replica->iofd;
 	event.events = EPOLLIN | EPOLLET;
 	rc = epoll_ctl(spec->receiver_epfd, EPOLL_CTL_ADD, iofd, &event);
 	if (rc == -1) {
 		return NULL;
```

Comment:
need to do a cleanup here of updating volstate again. Lets a add a 'TODO' here, and also a bug.
Similarly, for below cases also.

Reply:
ok.